src/Common/Tooltip/types.ts

```diff
@@ -0,0 +1,14 @@
+import { TippyProps } from '@tippyjs/react'
+
+export interface TooltipProps extends TippyProps {
+    /**
+     * If true, show tippy on truncate
+     * @default false
+     */
+    showOnTruncate?: boolean
+    /**
+     * If true, wrap with tippy irrespective of other options
+     * @default false
+     */
+    alwaysShowTippyOnHover?: boolean
+}
```

Comment:
```suggestion
type _TooltipProps =
    | {
          showOnTruncate?: boolean
          alwaysShowTippyOnHover?: never
      }
    | {
          /**
           * If true, show tippy on truncate
           * @default false
           */
          showOnTruncate?: never
          /**
           * If true, wrap with tippy irrespective of other options
           * @default false
           */
          alwaysShowTippyOnHover?: boolean
      }

export type TooltipProps = _TooltipProps & TippyProps
```

Reply:
Looks good

Reply:
Please follow naming convention for types we avoid starting naming with _

Reply:
Default values of types are misleading if both by default are false, then there is no meaning of wrapping this in a component since, that would only render text. The default behaviour should be to show tooltip on truncate

Reply:
I had asked @vivek-devtron what the default value for showOnTruncate should be. He had told me to make it false. What should I do here? @AbhishekA1509 @vivek-devtron

Reply:
Updated the name to `BaseTooltipProps`